backend/CircleForms/Services/IO/StaticFilesService.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.IO;
+using System.Threading.Tasks;
+using CircleForms.Models.Configurations;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace CircleForms.Services.IO;
+
+public class StaticFilesService : IStaticFilesService
+{
+    private readonly StaticFilesConfig _config;
+    private readonly ILogger<StaticFilesConfig> _logger;
+
+    public StaticFilesService(IOptions<StaticFilesConfig> config, ILogger<StaticFilesConfig> logger)
+    {
+        _logger = logger;
+        _config = config.Value;
+    }
+
+    public async Task WriteImageAsync(Stream image, string id, string filename)
+    {
+        var directory = Path.GetFullPath(Path.Combine(_config.VolumePath, id));
+        Directory.CreateDirectory(directory);
+
+        var filePath = Path.Combine(directory, filename);
+        _logger.LogDebug("Writing image to the static path {Path}", filename);
+        await using var fileStream = File.OpenWrite(filePath);
+        await image.CopyToAsync(fileStream);
```

Comment:
Are you sure http stream won't be closed & disposed by the time we get here? Maybe it's better to pass byte arrays around the async tasks instead of streams?

Reply:
yeah, i'm sure